sympy/solvers/solveset.py

```diff
@@ -882,6 +882,207 @@ def solveset_complex(f, symbol):
+    >>> x, y, z = symbols('x, y, z', real=True)
+    >>> sys = [x + y + z, x + y, z]
+    >>> eqs, eqs_dict = _sort_eqs(sys, [x, y, z])
+    >>> eqs
+    [z, x + y, x + y + z]
+    >>> eqs_dict
+    {z: set([z]), x + y: set([x, y]), x + y + z: set([x, y, z])}
+
+    """
+    from sympy.core.compatibility import ordered
+    _eq_unsolved_sym = {}
+    for eq in system:
+        _eq_unsolved_sym[eq] = _unsolved_syms(eq, all_symbols, known_symbols)
+
+    # sort such that equation with the fewest potential symbols is first.
+    # means eq with less variable first
+    eqs_sorted = list(
+        ordered(
+            system,
+            lambda _: len(_eq_unsolved_sym[_])
+        )
+    )
+    return eqs_sorted, _eq_unsolved_sym
+
+
+def eliminate(eqs, *symbols):
+    """
+
+    Eliminate the `symbols` from the equations and returns equation(s)
+    which does not contains `symbols`.
+
+    Parameters
+    ==========
+
+    eqs : list of equations
+        The target system of equations
+    symbols : list of Symbols
+        symbols to be eliminated
+
+    Raises
+    ======
+
+    ValueError
+        The input is not valid.
+    NotImplementedError
+        The value for symbol can not be substituted
+
+    Returns
+    =======
+
+    A FiniteSet of equations, does not contains `symbols`. If no such equation
+    is found then EmptySet.
+
+    Examples
+    ========
+
+    >>> from sympy.core.symbol import symbols
+    >>> from sympy.solvers.solveset import eliminate
+    >>> x, y, z = symbols('x, y, z', real=True)
+    >>> eliminate([x + y + z , y - z], y)
+    {-x - 2*z}
+    >>> eliminate([x + y + z , y - z, x - z], y)
+    {-x - 2*z, x - z}
+    >>> eliminate([x + y + z , y - z, x - y], y, z)
+    {3*x}
+    >>> eliminate([x + y + z], z, y)
+    EmptySet()
+
+    """
+    # It is implemented for domain=S.Reals
+    if not eqs:
+        return S.EmptySet
+
+    if not symbols:
+        return FiniteSet(*[eq for eq in eqs])
+
+    if hasattr(symbols[0], '__iter__'):
+        symbols = symbols[0]
+
+    try:
+        sym = symbols[0].is_Symbol
+    except AttributeError:
+        sym = False
+
+    if not sym:
+        msg = 'Symbols or iterable of symbols must be given as \
+        second argument, not type %s: %s'
+        raise ValueError(msg % (type(symbols[0]), symbols[0]))
+
+    res = []
+    for eq in eqs:
+        if simplify(eq).is_number:
+            eqs.remove(eq)
+        elif not eq.has(*symbols):
+            res.append(eq)
+            eqs.remove(eq)
+
+    if len(eqs) == 1:
+        eqs = simplify(eqs[0])
+        if not eqs.has(*symbols):
+            return FiniteSet(eqs)
+        else:
+            return S.EmptySet
+
+    eqs_new, eq_dict = _sort_eqs(eqs, symbols)
+    for eq in eqs_new[1:]:
+        eliminate_sym = eq_dict[eq]
+        eqs_used = dict(list(zip(eqs_new, [False for e in eqs_new])))
+        if len(eliminate_sym):
+            eq = simplify(eq.expand(force=True))
+            ind = eqs_new.index(eq)
```

Comment:
This line fails for me, since `eq` is now a different expression after expanding and simplification. For instance, when I run 
```
eliminate([-(x - 2*cos(t)**2)**2 - (y + 8*cos(t)**3)**2 - (z - 8*cos(t))**2 + 1, -8*(x - 2*cos(t)**2)*sin(t)*cos(t) + 48*(y + 8*cos(t)**3)*sin(t)*cos(t)**2 - 16*(z - 8*cos(t))*sin(t)], t)
```
I get the error:
```
ValueError: 8*(-x*cos(t) + 6*y*cos(t)**2 - 2*z + 48*cos(t)**5 + 2*cos(t)**3 + 16*cos(t))*sin(t) is not in list
```